### PR TITLE
[cucumber-react] Extract Cucumber-React to separate sub project

### PR DESCRIPTION
**Pull request by aslakhellesoy**
Extracting this to a separate npm module makes it possible to use the React component in other
projects (such as Cucumber Pro).

/cc @jbpros @romaingweb @mattwynne @tooky @sebrose any comments?

### CODE REVIEW

File: cucumber-react/test/reducer_test.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 /* eslint-env mocha */
 import assert from "assert"
-import reducer from "../../lib/common/reducer"
+import reducer from "../lib/reducer"
 
 describe(reducer.name, () => {
-  it("keeps a map of sources", () => {
+  it("keeps a map of parsed gherkin AST documents", () => {
```
Comment (mattwynne) [inline cucumber-react/test/reducer_test.js]: @aslakhellesoy when @tooky and I read the code we had to go looking to be sure that the `sources` in the state contained parsed ASTs and not raw source. Maybe `docs` or `gherkinDocs` would be a clearer name?

File: html-formatter/nodejs/lib/html_stream.jsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import Stream from "stream"
 import React from "react"
 import ReactDOMServer from "react-dom/server"
-import {Cucumber} from "./cucumber_react/cucumber_react"
+import {CucumberReact} from "cucumber-react"
+const {Cucumber} = CucumberReact
```
Comment (jbpros) [inline html-formatter/nodejs/lib/html_stream.jsx]: `import {Cucumber as CucumberReact} from "cucumber-react"`? This `const` trick is done in cucumber-react tests and possibly other places as well. It looks a bit odd.

File: html-formatter/nodejs/lib/react/html_formatter_app.jsx [hunk 3]
```diff
@@ -3,8 +3,8 @@ import React from "react"
 import {createStore} from "redux"
 import {connect, Provider} from "react-redux"
 import {render} from "react-dom"
-import reducer from "../common/reducer"
-import {Cucumber} from "../cucumber_react/cucumber_react"
+import {CucumberReact, reducer} from "cucumber-react"
+const {Cucumber} = CucumberReact
```
Comment (jbpros) [inline html-formatter/nodejs/lib/react/html_formatter_app.jsx]: `import {CucumberReact as Cucumber} ...`

File: html-formatter/nodejs/test/react_output.js [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import Stream from "stream"
-import React from "react" // eslint-disable-line no-unused-vars
+import React from "react"
 import {render} from "enzyme"
-import {Cucumber} from "../lib/cucumber_react/cucumber_react"
+import {CucumberReact} from "cucumber-react"
+const {Cucumber} = CucumberReact
```
Comment (jbpros) [inline html-formatter/nodejs/test/react_output.js]: `import { ... as ...}`

File: cucumber-react/test/cucumber_react_test.js [hunk 5]
```diff
@@ -2,8 +2,8 @@
 import assert from "assert"
 import React from "react"
 import {shallow} from "enzyme"
-import {GherkinDocument, Feature, Scenario, Step, Attachment} from "../../lib/cucumber_react/cucumber_react"
-import reducer from "../../lib/common/reducer" // eslint-disable-line no-unused-vars
+import {CucumberReact, reducer} from "../lib"
+const {GherkinDocument, Feature, Scenario, Step, Attachment} = CucumberReact
```
Comment (jbpros) [inline cucumber-react/test/cucumber_react_test.js]: I'd use two import statements.

**Comment by jbpros** [comment]
We (@romaingweb and me) made [another branch](https://github.com/cucumber/cucumber/tree/cucumber-react_cucumber-html-idiomatic-imports) , built on top of the one here, that suggests [improvements in the `import`s](https://github.com/cucumber/cucumber/compare/cucumber-react...cucumber-react_cucumber-html-idiomatic-imports). We were not sure if we should push the commit to the current PR, make a new one or ...?

**Pull request merged**